src/First/README.md

```diff
@@ -0,0 +1,451 @@
+# First
+
+```haskell
+First a = First (Maybe a)
+```
+
+`First` is a `Monoid` that will always return the first, non-empty value when
+(2) `First` instances are combined. `First` is able to be a `Monoid` because
+it implements a `Maybe` under the hood. The use of the `Maybe` allows for an
+empty to be represented with a `Nothing`.
+
+`First` can be constructed with either a value or a `Maybe` instance. Any value
+passed to the constructor will be wrapped in a `Just` to represent a non-empty
+instance of `First`. Any `Maybe` passed to the constructor will be lifted as
+is, allowing the ability to "choose" an value based on some disjunction.
+
+While most `Monoid`s only provide a [`valueOf`](#valueof) function used for
+extraction, `Last` takes advantage of its underlying `Maybe` to provide an
+additional [`option`](#option) method. Using [`valueOf`](#valueof) will extract
+the underlying `Maybe`, while [`option`](#option) will extract the underlying
+value in the `Maybe`, using the provided default value when the underlying
+`Maybe` is a `Nothing` instance.
+
+```javascript
+const First = require('crocks/First')
+
+const and = require('crocks/logic/and')
+const isNumber = require('crocks/predicates/isNumber')
+const mconcatMap = require('crocks/helpers/mconcatMap')
+const safe = require('crocks/Maybe/safe')
+
+// isEven :: Number -> Boolean
+const isEven =
+  x => !(x % 2)
+
+// isValid :: a -> Boolean
+const isValid =
+  and(isNumber, isEven)
+
+// chooseFirst :: [ * ] -> First Number
+const chooseFirst =
+  mconcatMap(First, safe(isValid))
+
+chooseFirst([ 21, 45, 2, 22, 19 ])
+  .valueOf(0)
+//=> Just 2
```

Comment:
nice example

Reply:
Except it is misleading, `valueOf` takes no arguments

Reply:
I WAS WONDERING ABOUT THAT. I mean the example as a whole 🤕